### ISSUE TITLE
write error: Invalid argument

### ISSUE THREAD

**Issue by ale5000-git** [rmyorston/busybox-w32#99]
This `cat /dev/urandom | tr -dc _A-Z-a-z-0-9 | head -c${1:-32};echo;`

give `tr: write error: Invalid argument` and `cat: write error: Invalid argument`.

**Comment by rmyorston** [comment]
The ```head``` command in the pipeline above only accepts a limited amount of input. On Unix when a process tries to write to a broken pipe it gets a SIGPIPE signal which, by default, will terminate the process. On Unix, therefore, the ```cat``` and ```tr``` commands terminate silently.

Microsoft Windows doesn't support SIGPIPE. Processes earlier in the broken pipeline have to handle a failed ```write``` function call. Often this results in the process seeing an unhelpful EINVAL error rather than the more useful EPIPE.

I recently made some changes to address this issue. The processes still don't fail silently but at least they report a meaningful 'Broken pipe' error.

However, these changes don't help in the above case because:
- I haven't released new binaries yet;
- the changes didn't handle the low-level ```write(2)``` function used by ```cat``` and ```tr```.

I've now added support for broken pipes to ```write(2)```.
